pkg/sinks/kafka/kafka.go

```diff
@@ -65,7 +67,7 @@ func NewToKafka(vertex *dfv1.Vertex, fromBuffer isb.BufferReader, opts ...Option
 			forwardOpts = append(forwardOpts, forward.WithReadBatchSize(int64(*x.ReadBatchSize)))
 		}
 	}
-	f, err := forward.NewInterStepDataForward(vertex, fromBuffer, map[string]isb.BufferWriter{vertex.Name: toKafka}, forward.All, applier.Terminal, nil, nil, forwardOpts...)
+	f, err := forward.NewInterStepDataForward(vertex, fromBuffer, map[string]isb.BufferWriter{generic.GetSinkOutboundEdge(vertex): toKafka}, forward.All, applier.Terminal, nil, nil, forwardOpts...)
```

Comment:
Can we get the buffer name from the Vertex object directly instead of adding an extra function `GetSinkOutboundEdge`?

Reply:
I suggested this because tomorrow, we might have different logic for each sink's outbound buffer.

Reply:
If this is the case, should we change the logic in

```
func GenerateSinkBufferName(namespace, pipelineName, vertex string) string {
	return fmt.Sprintf("%s-%s-%s_SINK", namespace, pipelineName, vertex)
}
```

Today we have several different things for edge/bugger:

1. Buffer name (includes edge buffer, source bugger and sink buffer), which is open to external;
2. Stream name (generated from buffer name), it's for internal stream/bucket creation.

`GetSinkOutboundEdge` introduces another thing open to external, and it's generated from buffer name.

Reply:
BTW - in `publishWatermark`, the map key is the sink buffer name, which is not consistent. 

Also, it's `nil, nil` here in this line.

Reply:
@whynowy Which key are you referring to?
Rest is done

Reply:
use this `vertex.GetToBuffers()[0].Name`